src/components/common/CountDownTimer.vue

```diff
@@ -0,0 +1,65 @@
+<script>
+import VueTimers from 'vue-timers/mixin';
+
+export default {
+  name: 'CountDownTimer',
+
+  props: {
+    timeout: {
+      type: Number,
+      default: 30000,
+    },
+    isLocked: {
+      type: Boolean,
+      default: false,
+    },
+    counter: {
+      type: Number,
+      default: 0,
+    },
+  },
+
+  watch: {
+    isLocked(newValue) {
+      if (!newValue) {
+        return;
+      }
+
+      this.startRequestTimer();
+    },
+  },
+
+  methods: {
+    startRequestTimer() {
+      this.$emit('update:is-locked', true);
+      this.$emit('update:counter', this.timeout / 1000);
```

Comment:
`update:counter` emitted in two points. 
for this point it's not so easy to understand that this is initiate `counter` value of property.
maybe change it to
```
const initialCounterValue = this.timeout / 1000;
this.$emit('update:counter', initialCounterValue);
```

or move emit to separated method for more readable
```
updateCounter(newValue) {
  this.$emit('update:counter', newValue);
}
```

Reply:
Usually we don't use namings like that. But I agree and support this idea.

Reply:
@dudiq @ifiri for future, we have a method name `startRequestTimer`. We don't need extra methods for one line `updateCounter`. If we go deeper we will add a lot of unnecessary methods for simple operations. If use this approach we will need creating `updateLockStatus`. It's ugly. `startRequestTimer` is enough. But `initialCounterValue` as computed is a good idea for this case